src/components/organize/tasks/forms/TaskDetailsForm.tsx

```diff
@@ -46,6 +47,22 @@ const TaskDetailsForm = ({ onSubmit, onCancel, task }: TaskDetailsFormProps): JS
         if (!values.campaign_id) {
             errors.campaign_id = intl.formatMessage({ id: 'misc.formDialog.required' });
         }
+        if (values.type === TASK_TYPE.VISIT_LINK || values.type === TASK_TYPE.SHARE_LINK) {
+            const config = values.config as VisitLinkConfig;
+            if (config?.url) {
+                const valid = validator.isURL(config.url);
+                if (!valid) {
+                    errors.config = {
+                        url: intl.formatMessage({ id: 'misc.formDialog.invalidUrl' }),
+                    };
```

Comment:
This solution is great in this very specific context. The problem however is that if there are any other errors to fields in the `config`, those will have to be checked after this, and that can be difficult to notice for a developer.

What happens if you add validation for another `config` attribute (e.g. the `fields` attribute for `demographic` tasks) before line 50?

The `errors.config = { /* … */ }` on line 55 would replace any errors set by that new validation, and those errors would be obfuscated.

So creating a new object like this is prone to create bugs down the line. Instead, you should only create a new object if `errors.config` is undefined. If it's defined, you don't need to create a new object before setting `errors.config.url`.

Reply:
Thanks for the clear guidance. I defined the `errors.config = {}` outside the `if statement`. Hope it's the ideal way of doing it.